docker/test_env/ethereum_env.go

```diff
@@ -108,265 +127,363 @@ func (b *EthereumNetworkBuilder) WithTest(t *testing.T) *EthereumNetworkBuilder
+	b.ethereumChainConfig = b.existingConfig.ethereumChainConfig
 }
 
 func (b *EthereumNetworkBuilder) validate() error {
 	if b.consensusType == nil {
 		return errors.New("consensus type is required")
 	}
-	if b.executionLayer == nil {
-		return errors.New("execution layer is required")
-	}
 
-	if *b.consensusType == ConsensusType_PoS && b.consensusLayer == nil {
-		return errors.New("consensus layer is required for PoS")
+	if len(b.participants) != 1 {
+		return errors.New("only one participant is currently supported")
 	}
 
-	if *b.consensusType == ConsensusType_PoW && b.consensusLayer != nil {
-		return errors.New("consensus layer is not allowed for PoW")
-	}
-
-	if b.consensusNodes > 1 {
-		return errors.New("only one consensus node is currently supported")
-	}
+	for _, p := range b.participants {
+		if p.ExecutionLayer == "" {
+			return errors.New("execution layer is required")
+		}
 
-	if b.executionNodes > 1 {
-		return errors.New("only one execution node is currently supported")
-	}
+		if *b.consensusType == ConsensusType_PoS && p.ConsensusLayer == "" {
+			return errors.New("consensus layer is required for PoS")
+		}
 
-	if *b.consensusType == ConsensusType_PoW && b.beaconChainConfig != nil {
-		return errors.New("beacon chain config is not allowed for PoW")
+		if *b.consensusType == ConsensusType_PoW && p.ConsensusLayer != "" {
+			return errors.New("consensus layer is not allowed for PoW")
+		}
 	}
 
-	if *b.consensusType == ConsensusType_PoW {
-		b.consensusNodes = 0
-	}
+	//TODO when we support multiple participants, we need to validate that all of them are either PoW or PoS
 
 	for _, addr := range b.addressesToFund {
 		if !common.IsHexAddress(addr) {
 			return fmt.Errorf("address %s is not a valid hex address", addr)
 		}
 	}
 
+	err := b.ethereumChainConfig.Validate(logging.GetTestLogger(b.t))
+	if err != nil {
+		return err
+	}
+
 	return nil
 }
 
-func (b *EthereumNetwork) startPos() (blockchain.EVMNetwork, RpcProvider, error) {
-	if b.ConsensusLayer != ConsensusLayer_Prysm {
-		return blockchain.EVMNetwork{}, RpcProvider{}, fmt.Errorf("unsupported consensus layer: %s", b.ConsensusLayer)
-	}
-	networkNames, err := b.getOrCreateDockerNetworks()
-	if err != nil {
-		return blockchain.EVMNetwork{}, RpcProvider{}, err
+type EthereumNetwork struct {
+	ConsensusType       ConsensusType                `json:"consensus_type"`
+	DockerNetworkNames  []string                     `json:"docker_network_names"`
+	Containers          EthereumNetworkContainers    `json:"containers"`
+	Participants        []EthereumNetworkParticipant `json:"participants"`
+	WaitForFinalization bool                         `json:"wait_for_finalization"`
+	isRecreated         bool
+	ethereumChainConfig *EthereumChainConfig
+	t                   *testing.T
+}
+
+func (en *EthereumNetwork) Start() (blockchain.EVMNetwork, RpcProvider, error) {
+	switch en.ConsensusType {
+	case ConsensusType_PoS:
+		return en.startPos()
+	case ConsensusType_PoW:
```

Comment:
Are there some tests that we still need this for? Or is this just to help along the transition?

Reply:
I didn't want to break backward-compatibility. Also, PoS is slower than PoW so some tests might fail because of that and we might to add some more checks/fluent waits.